rotkehlchen/chain/ethereum/modules/eigenlayer/balances.py

```diff
@@ -0,0 +1,125 @@
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.errors.misc import NotERC20Conformant, RemoteError
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+from .constants import CPT_EIGENLAYER
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import CHAIN_IDS_WITH_BALANCE_PROTOCOLS
+
+
+UNDERLYING_BALANCES_ABI: Final = [{'inputs': [], 'name': 'underlyingToken', 'outputs': [{'internalType': 'contract IERC20', 'name': '', 'type': 'address'}], 'stateMutability': 'view', 'type': 'function'}, {'inputs': [{'internalType': 'address', 'name': 'user', 'type': 'address'}], 'name': 'userUnderlyingView', 'outputs': [{'internalType': 'uint256', 'name': '', 'type': 'uint256'}], 'stateMutability': 'view', 'type': 'function'}]  # noqa: E501
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _read_underlying_assets(
+        evm_inquirer: 'EthereumInquirer',
+        strategy_address: ChecksumEvmAddress,
+        depositor: ChecksumEvmAddress,
+) -> tuple['FVal', 'EvmToken']:
+    """
+    Query the amount deposited in a eigenlayer strategy and the token of the strategy
+    May raise:
+    - RemoteError
+    - NotERC20Conformant
+    """
+    contract = EvmContract(
+        address=strategy_address,
+        abi=UNDERLYING_BALANCES_ABI,
+        deployed_block=0,
+    )
+    deposited_amount_raw = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='userUnderlyingView',
+        arguments=[depositor],
+    )
+    underlying_token_address = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='underlyingToken',
+    )
+    token = get_or_create_evm_token(
+        userdb=evm_inquirer.database,
+        evm_address=underlying_token_address,
+        chain_id=evm_inquirer.chain_id,
+    )
+    amount = asset_normalized_value(
+        amount=deposited_amount_raw,
+        asset=token,
+    )
+    return amount, token
+
+
+class EigenlayerBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EthereumInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
```

Comment:
why are you passing the chain id here? Is it supposed to be multichain?

Reply:
It is required by the interface but I believ it is not needed since it is stored in the inquirer